core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala

```diff
@@ -145,7 +147,7 @@ object SparkSubmit {
       }
     // In all other modes, just run the main class as prepared
     } else {
-      runMain(childArgs, childClasspath, sysProps, childMainClass)
+      runMain(childArgs, childClasspath, sysProps, childMainClass, args)
```

Comment:
Could we avoid having to pass `SparkSubmitArguments` to `runMain` and instead just deal with the `doAs` block out here? You only need it do it for this lower case. In this script there is a actually a fairly good separation of concerns here that the interface to `runMain` occurs strictly downstream from the somewhat complicated logic in `SparkSubmitArguments`.